site/app/templates/admin/admin_gradeable/AdminGradeablePeers.twig

```diff
@@ -141,36 +169,157 @@
-                });
-                if (grader_found == false) {
-                    //the entire grader row is removed from new peers
-                    grader_content.parents('tr').remove();
-                } else {
-                    //otherwise, fill in new students in the table
-                    let students_list = grader_content.parents('tr').find('.peer-table-students');
-                    students_list.empty();
-                    for (i in new_peers[grader_id]) {
-                        students_list.append('<span>'+new_peers[grader_id][i]+'</span> ');
-                    }
-                }
+                window.location.reload(false);
             },
-            error: function() {
+            error: function(err) {
+                console.error(err);
+                window.alert("Something went wrong. Please try again.");
+            }
+        });
+    }
+
+    /**
+    * Creates the popup for adding a new grader to the peer matrix. Triggered when
+    * the user presses the "Add New Grader to Matrix" button.
+    */
+    function addGraderPeerForm() {
+        // make add grader popup appear
+        $('.popup-form').css('display', 'none');
+        let form = $("#admin-gradeable-peers-form2");
+        form.css("display", "block");
+        $('#admin-gradeable-peers2-submit').prop('disabled', true);
+        $('#submit-error2').text("Please enter a valid new grader that is not already in the matrix.");
+        captureTabInModal("admin-gradeable-peers-form2");
+        // set the left panel to allow a new grader to be added
+        let peer_grader_pairs = JSON.parse($('#peer-grader-pairs').val());
+        let add_grader_div = $("#new-grader");
+        add_grader_div.empty();
+        add_grader_div.append("Grader to Add:<br />")
+        add_grader_div.append('<input type="text" id="new_peer_grader" name="new_peer_grader" /><br>');
+        let student_autofill = JSON.parse($('#student-autofill').val());
+        $('[name="new_peer_grader"]', form).autocomplete({
+            source: student_autofill
+        });
+        $('[name="new_peer_grader"]').autocomplete( "option", "appendTo", form ); 
+        // set the right panel to allow adding students for the new grader
+        let grader_students_div = $("#new-grader-students");
+        grader_students_div.empty();
+        grader_students_div.append("Students for New Grader:<br />")
+        grader_students_div.append('<input class="add_user_ids" type="text" name="add_user_id_0" /><br>');
+        $('[name="add_user_id_0"]', form).autocomplete({
+            source: student_autofill
+        });
+        $('[name="add_user_id_0"]').autocomplete( "option", "appendTo", form );
+        grader_students_div.append('<p id="add_student_button" style="cursor: pointer;" onclick="addStudentTextbox(\'#admin-gradeable-peers-form2\',\'#new-grader-students\');" aria-label="Add More Users"><i class="fas fa-plus-square"></i>Add More Users<br></p>');
```

Comment:
I think that only `input` element should be added via JS  and rest of the content should be part of twig file itself... you can wrap the input element with a div and then use `append/prepend` on that container to dynamically change the number of input elements ... same comment for `adminGradeablePeerForm`

Reply:
Oops I just realized my comments for Shail from yesterday didn't post. But I have the same reply, unless I'm misunderstanding something: "Hmm so it's for a popup, and I did it dynamically because every time the popup starts again I wanted all the input fields to be emptied out again, and the number of text input fields available to go back to the default (there's a +Add New Student button that allows them to add new text boxes)."